### ISSUE TITLE
Ports, ports, and more ports

### ISSUE THREAD

**Issue by iconoclasthero** [krisek/audioloader#12]
1.  Need a way to change the default server daemon port from 6600 to 6602 without having to use the web interface at all (i.e., needs to be done on the back end so user doesn't have to do anything)
~~2. Need to change the default web port from 5000; I already have kavita on 5000~~ > In start-u.py, change 5000 to whatever.
3. What is 3400 being used for?

**Comment by iconoclasthero** [comment]
I've literally changed every instance of 6600 I could find in the git clone directory and it still defaults to 6600.  Also, there's no place to put a password in on the web interface

```
~/bin/gits/audioloader $ find . -type f -exec grep -H 6600 "{}" \;
grep: ./venv/bin/uwsgi: binary file matches
./venv/lib/python3.10/site-packages/pip/_vendor/idna/idnadata.py:        0x1d6600001d6b,
./venv/lib/python3.10/site-packages/pip/_vendor/idna/idnadata.py:        0xff660000ff70,
./venv/lib/python3.10/site-packages/pip/_vendor/idna/idnadata.py:        0x96600000970,
./venv/lib/python3.10/site-packages/pip/_vendor/idna/idnadata.py:        0xa6600000a76,
./venv/lib/python3.10/site-packages/pip/_vendor/idna/idnadata.py:        0xb6600000b70,
./venv/lib/python3.10/site-packages/pip/_vendor/idna/idnadata.py:        0xc6600000c70,
./venv/lib/python3.10/site-packages/pip/_vendor/idna/idnadata.py:        0xd6600000d70,
./venv/lib/python3.10/site-packages/pip/_vendor/idna/idnadata.py:        0x1106600011076,
./venv/lib/python3.10/site-packages/pip/_vendor/idna/idnadata.py:        0x113660001136d,
./venv/lib/python3.10/site-packages/pip/_vendor/idna/idnadata.py:        0x6600000066a,
./venv/lib/python3.10/site-packages/idna/idnadata.py:        0x1d6600001d6b,
./venv/lib/python3.10/site-packages/idna/idnadata.py:        0xff660000ff70,
./venv/lib/python3.10/site-packages/idna/idnadata.py:        0x96600000970,
./venv/lib/python3.10/site-packages/idna/idnadata.py:        0xa6600000a76,
./venv/lib/python3.10/site-packages/idna/idnadata.py:        0xb6600000b70,
./venv/lib/python3.10/site-packages/idna/idnadata.py:        0xc6600000c70,
./venv/lib/python3.10/site-packages/idna/idnadata.py:        0xd6600000d70,
./venv/lib/python3.10/site-packages/idna/idnadata.py:        0x1106600011076,
./venv/lib/python3.10/site-packages/idna/idnadata.py:        0x113660001136d,
./venv/lib/python3.10/site-packages/idna/idnadata.py:        0x6600000066a,
./venv/lib/python3.10/site-packages/charset_normalizer/constant.py:    "Ancient Symbols": range(65936, 66000),
./venv/lib/python3.10/site-packages/charset_normalizer/constant.py:    "Phaistos Disc": range(66000, 66048),

```

**Comment by krisek** [comment]
There are three (2+1) services running:

- mpd, normally on port 6600, this is the default along the app, I wouldn't think about changing it
- flask backend, this is on port 5000 if you don't use any wsgi environment, and I use port 3400 for wsgi. 
- web server port (80 and/or 443) in front of wsgi

I encourage you to use gunicorn as wsgi server, as uwsgi turns out to be a bit complicated on alternative architectures. Maybe I should remove `start-u.sh` and `start-uwsgi.sh` from the repo if they cause confusion.

The mpd port is a client setting (the flask code is merely a proxy to mpd), it should be straightforward to change there. If you want to change it in flask, then you need to adjust the code: a replace all on `routes.py` should do the job. (Maybe I can put this into a variable, but really, I don't see the use case, why to configure this in flask.)

On this screenshot you can see how to change the mpd server port.

![image](https://github.com/krisek/audioloader/assets/786044/b492a42a-26b2-4ef3-b3aa-b8dbc4b69bb0)

And yes, before I forget :) for the sake of total confusion, if you use mpd http audio output, then you have a fourth port in the picture.

I know it is complex with such a modular architecture... I'll try to answer if you have further questions.

**Comment by iconoclasthero** [comment]
Yes, well in this case I have someone with macular degeneration who I want to be able to serve audiobooks and let her hit pause/play.  6600 is in use for my primary mpd instance (which I'm listening to the Grateful Dead on right now), and I made a second server instance that is controlled via 6602.  I changed every single 6600 in the git repo directory to 6602 including ./audioloader/app/routes.py and it still goes to 6600 by default.

I'm actually just grabbing my icecast2 stream since I don't have the httpd set up in mpd, regardless, the piece I need to fix is the 6600 by default so I don't have to explain to an 80 yo how to change the port.

**Comment by krisek** [comment]
Have you tried to set the mpd port on client side, as my screenshot shows? Basically, there is no need to change anything in the code. 

TBH, I just realized after reviewing the code, that even if you replace all 6600 in routes.py the frontend/client will still use/enforce port 6600, until you don't configure it otherwise. Apologies for the misunderstanding -- I wrote this code a few years back, I don't remember everything.

So just go to the web interface, click the gear icon, and set mpd port 6602, it should work.

Sorry for the misunderstanding again.
